db/system_distributed_keyspace.cc

```diff
@@ -292,26 +292,26 @@ static cdc::token_range_description get_token_range_description_from_value(const
 
 future<>
 system_distributed_keyspace::insert_cdc_topology_description(
-        db_clock::time_point time,
+        cdc::generation_id gen_id,
         const cdc::topology_description& description,
         context ctx) {
     return _qp.execute_internal(
             format("INSERT INTO {}.{} (time, description) VALUES (?,?)", NAME, CDC_TOPOLOGY_DESCRIPTION),
             quorum_if_many(ctx.num_token_owners),
             internal_distributed_query_state(),
-            { time, make_list_value(cdc_generation_description_type, prepare_cdc_generation_description(description)) },
+            { gen_id.ts, make_list_value(cdc_generation_description_type, prepare_cdc_generation_description(description)) },
```

Comment:
I think this is breaching the abstraction. One value of introducing the type is ability to hide its implementation. Let's make ts private and use get_ts everywhere - shall we?

Reply:
Using `get_ts` here would assume that the generation ID is only consisting of the timestamp, but that doesn't have to be true (and it won't be true in the future - hence the refactor). The fact that currently all generation IDs consist only of the timestamp is an implementation detail. `insert_cdc_topology_description` is serializing the generation ID. In order to serialize it, it must know its internals - the full definition of the struct.

Summarizing, using `get_ts` here would be breaking abstraction. Not using it here is the correct usage of the generation ID abstraction.